src/wrappers/amqplib.js

```diff
@@ -0,0 +1,134 @@
+
+/**
+ * acts as a middleware for `consumer.run()`
+ * @param {object} message the messages param to send
+ * @param {Function} callback the callback function
+ * @param {Channel} channel the Channel object of amqplib
+ * @param {function} originalHandler original consumer function
+ */
+function amqplibSubscriberMiddleware(message, callback, channel) {
+    let originalHandlerSyncErr;
+    try {
+        // Initialize tracer and runner.
+        tracer.restart();
+        const { slsEvent: amqpEvent, startTime: amqpStartTime } =
+        eventInterface.initializeEvent(
+            'rabbitmq',
+            message.fields.routingKey,
+            'consume',
+            'trigger'
+        );
+
+        const metadata = {
+            exchange: message.fields.exchange,
+            redelivered: message.fields.redelivered,
+            host: channel.connection.stream._host, // eslint-disable-line no-underscore-dangle
+            consumer_tag: message.fields.consumerTag,
+        };
+        if (message.properties.headers[EPSAGON_HEADER]) {
+            metadata[EPSAGON_HEADER] = message.properties.headers[EPSAGON_HEADER].toString();
+        }
+
+        tracer.addEvent(amqpEvent);
+        eventInterface.finalizeEvent(amqpEvent, amqpStartTime, null, metadata, {
+            headers: message.properties.headers,
+            message: message.content.toString(),
+        });
+
+        const { label, setError } = tracer;
+        // eslint-disable-next-line no-param-reassign
+        message.epsagon = {
+            label,
+            setError,
+        };
+        const { slsEvent: nodeEvent, startTime: nodeStartTime } = eventInterface.initializeEvent(
+            'node_function', 'message_handler', 'execute', 'runner'
+        );
+        let runnerResult;
+        try {
+            runnerResult = callback(message);
+        } catch (err) {
+            originalHandlerSyncErr = err;
+        }
+
+        if (callback.name) {
+            nodeEvent.getResource().setName(callback.name);
+        }
+
+        // Handle and finalize async user function.
+        if (utils.isPromise(runnerResult)) {
+            let originalHandlerAsyncError;
+            runnerResult = runnerResult.catch((err) => {
+                originalHandlerAsyncError = err;
+                throw err;
+            }).finally(() => {
+                eventInterface.finalizeEvent(nodeEvent, nodeStartTime, originalHandlerAsyncError);
+                tracer.sendTrace(() => {});
+            });
+        } else {
+            // Finalize sync user function.
+            eventInterface.finalizeEvent(nodeEvent, nodeStartTime, originalHandlerSyncErr);
+            tracer.sendTrace(() => {});
+        }
+        tracer.addRunner(nodeEvent, runnerResult);
+    } catch (err) {
+        tracer.addException(err);
+    }
+    // Throwing error in case of sync user function.
+    if (originalHandlerSyncErr) {
+        throw originalHandlerSyncErr;
+    }
```

Comment:
you forgot to return `runnerResult` here

Reply:
👍